lib/domain/token.js

```diff
@@ -1,9 +1,10 @@
 import ArgumentError from './argument-error';
 
 export default class Token {
-  constructor({ token, createdDate }) {
+  constructor({ token, createdDate, shared }) {
     if (!token) throw new ArgumentError('no token');
     this.token = token;
     this.createdDate = createdDate || new Date(Date.now()).toISOString();
+    this.shared = shared || false;
```

Comment:
I think we need to change the token code a little - have "shared" as a getter (see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get) that looks at the date, and change createdDate to sharedDate. atm createdDate is set when the token is created but we actually dont want to set the date til it's been shared